perception/mil_vision/src/pc_colorizer.cpp

```diff
@@ -0,0 +1,147 @@
+#include <ros/ros.h>
+#include <deque>
+
+#include <sensor_msgs/PointCloud2.h>
+
+#include <mil_vision_lib/image_acquisition/ros_camera_stream.hpp>
+
+#include <geometry_msgs/TransformStamped.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <pcl/PCLPointCloud2.h>
+#include <pcl/conversions.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl_ros/point_cloud.h>
+#include <pcl_ros/transforms.h>
+
+#include <opencv2/opencv.hpp>
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+using ROSCameraStream_Vec3 = mil_vision::ROSCameraStream<cv::Vec3b>;
+//! Point Cloud Colorizer class.
+/*! Subscribes to cameras and velodyne topic to publish a RGB pointcloud.
+  Inputs: rosparam '/pc_colorizer/cameras', and topic '/velodyne_points'
+  Outputs: topic '/velodyne_points/colored'
+*/
+class Colorizer
+{
+  ros::NodeHandle nh_;
+  //! Deque to store camera streams for each camera
+  std::deque<std::unique_ptr<ROSCameraStream_Vec3>> cameras_;
+
+  ros::Subscriber point_cloud_sub_;
+  ros::Publisher rgb_cloud_pub_;
+
+  tf2_ros::Buffer tf_buffer_;
+  tf2_ros::TransformListener tf_listener_;
+
+public:
+  //! Initialize tf buffer and resolve camera feeds
+  Colorizer() : nh_(), tf_listener_(tf_buffer_)
+  {
+    std::vector<std::string> camera_names = { "/stereo/right/image_rect_color", "/camera/front/right/"
+                                                                                "image_rect_color" };
```

Comment:
What kind of wizardry is this? Didn't know this is valid syntax

Reply:
c++11 list initialization is a beautiful thing

Reply:
yeah yeah yeah but look closely at it... there's 3 strings

Reply:
Also just realized it should probaby default to /camera/front/right/image_rect_color and /camera/front/left/image_rect_color

Reply:
I think this is an unintended result in clang format. Might be a bug. Dunno. I added an extra camera topic so the clang format shifted all the strings to their own lines. Interesting....